raiden/api/rest.py

```diff
@@ -2048,22 +2055,25 @@ def register_light_client(self,
             signed_password,
             password,
             signed_display_name,
-            signed_seed_retry)
-
-        if light_client is not None and light_client["result_code"] == 200:
+            signed_seed_retry
+        )
 
-            matrix_light_client_transport_instance = get_matrix_light_client_instance(
+        if light_client and light_client["result_code"] == 200:
+            config = self.raiden_api.raiden.config["transport"]["matrix"]
+            config["light_client_password"] = light_client["encrypt_signed_password"]
+            config["light_client_display_name"] = light_client["encrypt_signed_display_name"]
+            config["light_client_seed_retry"] = light_client["encrypt_signed_seed_retry"]
+            light_client_transport = transport_config.transport_layer.new_light_client_transport(
                 address=light_client["address"],
-                config=self.raiden_api.raiden.config["transport"]["matrix"],
-                password=light_client["encrypt_signed_password"],
-                display_name=light_client["encrypt_signed_display_name"],
-                seed_retry=light_client["encrypt_signed_seed_retry"])
+                config=config
+            )
```

Comment:
this now puts all the needed params for new LC instances inside of the `config` dict, and uses the transport layer static method to create these new transport instances

Reply:
i dont understand this, on the config for matrix transport, you put in the "light_client_password" key, the password of a specific light client

every time we register a new one, the config.transport.matrix.light_client_passwordm config.transport.matrix.display_name etc will be changed

why we need this?

if it is only for the new_light_client_transport i suggest to pass another object, not to modify the config

Reply:
thanks, good point.

i've changed this so that the light client object receives an `auth_params` dictionary as an additional constructor parameter, and moved these variables out of the `config` param.